x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/AsyncSearchResponse.java

```diff
@@ -157,6 +157,10 @@ public long getExpirationTime() {
         return expirationTimeMillis;
     }
 
+    public void setExpirationTime(long expirationTimeMillis) {
+        this.expirationTimeMillis = expirationTimeMillis;
```

Comment:
I wonder if we could add a constructor that takes this, it seems possible given that we call decode in one place only and it could accept the expiration time too, and pass it to the constructor.

Reply:
+1, I pushed https://github.com/elastic/elasticsearch/pull/55435/commits/47f35443ad45f0fb2ac4959761c99ff195be22ac